tests/test_new_dataset.py

```
@@ -0,0 +1,287 @@
+    with h5py.File(store_path / f"{data.session.id}.h5", "w") as f:
+        data.to_hdf5(f, serialize_fn_map=serialize_fn_map)
+
+    return store_path
+
+
+class TestDataset:
+    def test_recording_discovery(self, dummy_spiking_brainset):
+        # When recording ids are not specified, root is a string
+        ds = Dataset(str(dummy_spiking_brainset))
+        expected_rec_ids = ["session1", "session2", "session3", "session4"]
+        for actual_id, expected_id in zip(ds.recording_ids, expected_rec_ids):
```

Ruff (0.14.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)